build.gradle

```diff
@@ -34,131 +33,7 @@ allprojects {
-    robolectric = "org.robolectric:robolectric:$robolectricVersion"
-
-    //supportLib
-    supportv4 = "androidx.legacy:legacy-support-v4:$support_version"
-    appCompat = "androidx.appcompat:appcompat:$appCompatVersion"
-    materialDesign = "com.google.android.material:material:$materialDesignVersion"
-    recyclerView = "androidx.recyclerview:recyclerview:$support_version"
-    cardView = "androidx.cardview:cardview:$support_version"
-    coreKtx = "androidx.core:core-ktx:$coreKtxVersion"
-
-    //rx
-    rxJava = "io.reactivex.rxjava2:rxjava:$rxJavaVersion"
-    rxAndroid = "io.reactivex.rxjava2:rxandroid:$rxAndroidVersion"
-    rxKotlin = "io.reactivex.rxjava2:rxkotlin:$rxKotlinVersion"
-    rxAnimations = "com.mikhaellopez:rxanimation:$rxAnimationsVersion"
-
-    //arch-comp Lifecycles, LiveData, and ViewModel
-    lifecycleExt = "androidx.lifecycle:lifecycle-extensions:$lifecycleVersion"
-    lifecycleAnnotation = "androidx.lifecycle:lifecycle-compiler:$lifecycleVersion"
-    liveDataKtx = "androidx.lifecycle:lifecycle-livedata-ktx:$lifecycleVersion"
-    lifecycleViewModelKtx = "androidx.lifecycle:lifecycle-viewmodel-ktx:$lifecycleViewModelKtxVersion"
-
-    //algolia
-    algolia = "com.algolia:algoliasearch-android:$algoliaVersion"
-
-    //navigation
-    navigationFragment = "androidx.navigation:navigation-fragment-ktx:$navigationVersion"
-    navigationUI = "androidx.navigation:navigation-ui-ktx:$navigationVersion"
-
-    //room
-    roomRuntime = "androidx.room:room-runtime:$roomVersion"
-    roomCompiler = "androidx.room:room-compiler:$roomVersion"
-
-    //di
-    hilt = "com.google.dagger:hilt-android:$hiltVersion"
-    hiltKapt = "com.google.dagger:hilt-compiler:$hiltVersion"
-
-    //net
-    retrofit = "com.squareup.retrofit2:retrofit:$retrofitVersion"
-    okHttp = "com.squareup.okhttp3:okhttp:$okHttpVersion"
-    rxJavaAdapter = "com.squareup.retrofit2:adapter-rxjava2:$retrofitVersion"
-
-    moshiConverter = "com.squareup.retrofit2:converter-moshi:$moshiConverterVersion"
-    moshi = "com.squareup.moshi:moshi:$moshiVersion"
-    moshiKotlin = "com.squareup.moshi:moshi-kotlin:$moshiVersion"
-    moshiKapt = "com.squareup.moshi:moshi-kotlin-codegen:$moshiVersion"
-
-    //picasso
-    picasso = "com.squareup.picasso:picasso:$picassoVersion"
-
-    //threetenabp
-    threetenabp = "com.jakewharton.threetenabp:threetenabp:$threetenabpVersion"
-
-    //shadowView
-    shadowView = "com.loopeer.lib:shadow:$shadowViewVersion"
-
-    //timber
-    timber = "com.jakewharton.timber:timber:$timberVersion"
-
-    //stetho
-    stethoCore = "com.facebook.stetho:stetho:$stethoVersion"
-    stethoOkHttp = "com.facebook.stetho:stetho-okhttp3:$stethoVersion"
-    stethoUrlConnection = "com.facebook.stetho:stetho-urlconnection:$stethoVersion"
-
-    //plugins
-    ktlint = "0.41.0"
-
     buildTime = new Date().format("yyyy-MM-dd'T'HH:mm'Z'", TimeZone.getTimeZone("UTC"))
```

Comment:
Nitpicking: This also can be moved to `buildSrc` and converted to an extension function and used by child `Gradle` scripts. If you want, you can do this change, too.

Reply:
I removed `ext { }` in `build.gradle(Weatherapp)` and moved to `Configs.kt`.

dabaec6